dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java

```diff
@@ -1953,4 +1958,36 @@ private boolean isNewProcessInstance() {
         }
     }
 
+    private void setGlobalParamIfCommanded(ProcessDefinition processDefinition, Map<String, String> cmdParam) {
+        // get start params from command param
```

Comment:
I don't see the whole code, It's better to check the `cmdParam` in the begin, if the `cmdParam` is null, you can directly return. You can simplify the logic.

Reply:
Map<String, String> cmdParam = JSONUtils.toMap(processInstance.getCommandParam());
            if (cmdParam != null && cmdParam.containsKey(CMDPARAM_COMPLEMENT_DATA_START_DATE)) {

Reply:
hi ,
the code already have judged null.
yours

Reply:
Yes, I mean you can just check once in the beginning.

Reply:
have done , thanx